content/blog/automation-api-as-platform/index.md

```diff
@@ -0,0 +1,275 @@
+---
+title: "Build Self-Service Cloud Infrastructure with Automation API "
+date: 2020-11-09
+meta_desc: "With Pulumi's Automation API, you can build a REST API for creating, reading, updating, and deleting (CRUD) cloud resources."
+meta_image: automation_api.png
+authors:
+    - sophia-parafina
+tags:
+    - Automation API
+    - CRUD
+---
+
+If you could create infrastructure without using a cloud provider's console, a CLI, or a templating engine, what would you build? Pulumi's Automation API lets you create declarative infrastructure defined by your best practices and expose it behind a REST, gRPC, or custom API.
+
+So just what is Automation API? Think of it as Pulumi's infrastructure as code engine as an SDK. Instead of writing code and using the CLI to declare infrastructure, you can directly tell the engine to build your infrastructure. Imagine building an application that creates infrastructure via a REST interface. Get ready, because that's what we're going to do.
+
+<!--more-->
+
+## Self-service Cloud Infrastructure
+
+Automation API is just another package that runs inside your favorite frameworks and works with other packages. You can create methods or functions to create infrastructure that you can call from other frameworks. One Automation API use case is self-service infrastructure. You can create declarative infrastructure and expose it behind a REST interface.
+
+This example demonstrates how to create infrastructure with Automation API and the [Express](https://expressjs.com/) Node.js  framework. To keep this example simple, we'll deploy the REST API locally. To deploy the code in the cloud, for example, in a virtual machine, we'll need to create an [instance profile](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html) to store AWS credentials. Deploying Automation API is beyond this article's scope, so we'll use `localhost` as the server's endpoint. We'll cover deploying Automation API in the cloud in subsequent articles.
+
+### Infrastructure as a function
+
+We use an `inline` Pulumi program that doesn't require a separate package with `index.ts` and `Pulumi.yaml` files. Inline programs are functions that can be authored in an `index.ts` file or be imported from another package. The example deploys an AWS S3 website with the context and deployment automation defined in a function.
```

Comment:
> The example deploys an AWS S3 website

Conceptually, our custom unit of infrastructure is a static S3 website in this example. Pulumi Automation API allows us to take our custom infrastructure component, and turn it into a higher level RESTful resource that can be managed programmatically. 

While the unit of custom infrastructure in this is example is a static website with content, it could just as easily be full web services, or more complex deployments like kubernetes clusters with accompanying docker images and AWS resources.

Going back to the previous paragraph:

> You can create methods or functions to create infrastructure that you can call from other frameworks.

I think we may want to consider discussing "components", "templates", or "platforms". There is a "thing" defined with pulumi, and the automation API allows you CRUD that thing dynamically.

Reply:
addressed in previous comment